### ISSUE TITLE
Validate config files

### ISSUE THREAD

**Issue by noib3** [zed-industries/feedback#101]
**Is your feature request related to a problem? Please describe.**

After reading the docs on the website I quickly wrote this basic `settings.json`:

```json
{
    "format-on-save": true,
    "preferred-line-length": 79,
    "soft-wrap": "preferred_line_length",
    "theme": "rosé-pine-dawn",
    "vim-mode": true,
}
```

but after saving the file none of the configs were applied. Looking at it a little bit closer I remembered that json is particularly anal about trailing commas. Ok, changing `"vim-mode": true,` to `"vim-mode": true` should fix it, right? Nope, still no luck.

About quitting and reopening Zed a few times I checked back the docs to make sure I didn't misspell any settings. Sure enough, all the dashes (`-`) were supposed to be underscores (`_`) :/

**Describe the solution you'd like**

Validating both the JSON syntax and the settings' names and values would definitely be a welcome addition. For example, both `"format-on-save": true` and `"format_on_save": 42` should cause the user to be shown an error message describing the issue.

Just as an example, here's what the Alacritty terminal shows when it tries to load an invalid config file:

<img width="1013" alt="Screen Shot 2022-06-20 at 6 26 51 PM" src="https://user-images.githubusercontent.com/59321248/174651177-924ec74f-8acf-4ded-9939-d31f0c32bb90.png">

Btw, if you're using Serde for the deserialization there's the [serde_path_to_error](https://docs.rs/serde_path_to_error/latest/serde_path_to_error/) crate which gives you the full path of a config option that caused a deserialization error.

**Comment by iamnbutler** [comment]
Dig it 👍🏽 We'd need to decide where to show it however.

**Comment by iamnbutler** [comment]
Also sorry about your experience with `-`/`_`. Was it not suggesting completions for you in `settings.json`?

**Comment by noib3** [comment]
> Was it not suggesting completions for you in settings.json?

Nope. Not sure if that's caused by the json lsp not getting installed successfully (don't have `npm` in my PATH). Shouldn't be.

**Comment by as-cii** [comment]
Yeah I believe that might be why you're not getting errors or suggestions in your JSON file. Can you try installing `npm` and seeing if that helps? Thanks!

**Comment by noib3** [comment]
Yep, I get completions if the lsp is running. Still not seeing syntax errors though

<img width="221" alt="Screen Shot 2022-06-21 at 10 56 07 AM" src="https://user-images.githubusercontent.com/59321248/174760122-a92ced18-2175-40f1-9490-1523a27538f8.png">
